UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -17,39 +17,68 @@ public class NMDataService : INMDataService
         private const string YearAndWeek = "YEAR/WEEK";
         private readonly IHttpClientFactory _httpClientFactory;
         private readonly IOptions<FileShareServiceConfiguration> _fileShareServiceConfig;
+        private readonly IFileShareServiceCache _fileShareServiceCache;
+        private readonly IOptions<CacheConfiguration> _cacheConfiguration;
 
-        public NMDataService(IFileShareService fileShareService, ILogger<NMDataService> logger, IAuthFssTokenProvider authFssTokenProvider, IHttpClientFactory httpClientFactory, IOptions<FileShareServiceConfiguration> fileShareServiceConfig)
+        public NMDataService(IFileShareService fileShareService, ILogger<NMDataService> logger, IAuthFssTokenProvider authFssTokenProvider, IHttpClientFactory httpClientFactory, IOptions<FileShareServiceConfiguration> fileShareServiceConfig,
+                             IFileShareServiceCache fileShareServiceCache, IOptions<CacheConfiguration> cacheConfiguration)
         {
             _fileShareService = fileShareService;
             _logger = logger;
             _authFssTokenProvider = authFssTokenProvider;
             _fileShareServiceConfig = fileShareServiceConfig;
             _httpClientFactory = httpClientFactory;
+            _fileShareServiceCache = fileShareServiceCache;
+            _cacheConfiguration = cacheConfiguration;
         }
 
-        public async Task<List<ShowFilesResponseModel>> GetWeeklyBatchFiles(int year, int week, string correlationId)
+        public async Task<ShowNMFilesResponseModel> GetWeeklyBatchFiles(int year, int week, string correlationId)
         {
             try
             {
+                BatchSearchResponse SearchResult = new();
+                BatchSearchResponseModel batchSearchResponseModel = new();
 
-                string accessToken = await _authFssTokenProvider.GenerateADAccessToken(correlationId);
+                if (_cacheConfiguration.Value.IsFssCacheEnabled)
+                {
+                    batchSearchResponseModel = await _fileShareServiceCache.GetWeeklyBatchFilesFromCache(year, week, correlationId);
+
+                    if (batchSearchResponseModel.batchSearchResponse != null) {
+                        SearchResult = batchSearchResponseModel.batchSearchResponse;
+                    }
+                }
+                
+                if (SearchResult.Entries == null)
```

Comment:
If there is an exception thrown in reading the cache I think this method will exit without returning data....?  Wouldn't it be better in that case to revert to calling the FSS?  Could you please confirm the programme flow for Exception scenarios using unit testing?